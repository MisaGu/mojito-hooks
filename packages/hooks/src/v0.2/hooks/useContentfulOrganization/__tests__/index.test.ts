import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulOrganizations,
  UseContentfulOrganizationsProps,
  UseContentfulOrganizationsReturn,
} from '../useContentfulOrganizations';

describe('useContentfulOrganizations()', () => {
  it('should be defined', () => {
    expect(useContentfulOrganizations).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulOrganizationsProps?],
      UseContentfulOrganizationsReturn
    >(
      (args = []) => {
        return useContentfulOrganizations(...args);
      },
      {
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
