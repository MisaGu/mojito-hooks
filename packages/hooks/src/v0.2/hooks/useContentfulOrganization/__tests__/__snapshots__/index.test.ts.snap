// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulOrganizations() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "organizations": Array [
    Object {
      "homepageRedirect": Object {
        "name": "Mock Collection",
        "slug": "mock-collection",
      },
    },
  ],
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": true,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
